iceoryx_utils/include/iceoryx_utils/cxx/timer.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_UTILS_PERIODIC_TIMER_HPP
```

Comment:
Okay I renamed to `periodic_timer` and moved to the `posix` namespace

Reply:
Please don't rename it to `periodic_timer`. I know from other timer implementations that they have a `oneShot` functionality. This could easily be added to the `Timer` itself, but not to a new class which wraps the `Timer`.

Reply:
@mossmaurice wants to have the file named as `periodic_timer`?

Reply:
What about the following:

1) For now we stick with `periodic_timer` and get this PR merged
2) We remove the old `posix::timer` and rename the new one to something we all like

In the beginning of this thread I was just noticing the difference between include guard and file name. I would be fine with `periodic_timer` as a final name.

Reply:
@mossmaurice Yes I agree with the above plan.

Initially I named this file as `timer` and after which i modified this to `periodic_timer` as per your feedback.